build/assets/tuned/openshift-node-performance

```diff
@@ -85,7 +85,7 @@ initrd_add_dir=
 # overrides cpu-partitioning cmdline
 cmdline_cpu_part=+nohz=on rcu_nocbs=${isolated_cores} tuned.non_isolcpus=${not_isolated_cpumask} intel_pstate=disable nosoftlockup
 {{if .StaticIsolation}}
-cmdline_realtime=+tsc=nowatchdog intel_iommu=on iommu=pt isolcpus=${isolated_cores} systemd.cpu_affinity=${not_isolated_cores_expanded}
+cmdline_realtime=+tsc=nowatchdog intel_iommu=on iommu=pt isolcpus=managed_irq,${isolated_cores} systemd.cpu_affinity=${not_isolated_cores_expanded}
```

Comment:
I think I need more context about what isolcpus=managed_irq does. Because old isolcpus disables cpu process balancing for those cores (isolated_cores). However irqs have impact even in cases when cpu balancing is still enabled and we need to redirect them to not_isolated_cores in all cases (even if static isolation is not wanted).

So it is very likely the managed_irq change is a bit more complicated than this.

Also make sure the option is only used in kernel versions that support that argument. Because it is rather new. You will probably need a conditional section for that (see https://github.com/redhat-performance/tuned/commit/8d9cd00387426ed0)

Reply:
I think we can use https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/commit/?id=11ea68f553e244851d15793a7fa33a97c46d8271 as context.

The way I understand it (I might be wrong) managed_irq flag works only in combination with a X-Y cpu range of isolcpus and aims to prevent the case where interrupts are routed to the X-Y range. Meaning, no range no case: ```This isolation is best effort and only effective if the automatically
assigned interrupt mask of a device queue contains isolated and
housekeeping CPUs. ```

Another interesting point is the solution is not bullet free:  ```If a queue's affinity mask contains only isolated CPUs then this parameter
has no effect on the interrupt routing decision...```

However I did find an issue, here is housekeeping_isolcpus_setup function:
 `while (isalpha(*str)) {
                if (!strncmp(str, "nohz,", 5)) {
                        str += 5;
                        flags |= HK_FLAG_TICK;
                        continue;
                }

                if (!strncmp(str, "domain,", 7)) {
                        str += 7;
                        flags |= HK_FLAG_DOMAIN;
                        continue;
                }

                if (!strncmp(str, "managed_irq,", 12)) {
                        str += 12;
                        flags |= HK_FLAG_MANAGED_IRQ;
                        continue;
                }

                //[...]
        }

        /* Default behaviour for isolcpus without flags */
        if (!flags)
                flags |= HK_FLAG_DOMAIN;

So if we add managed_irq flag we remove the main isolation logic, so we need to add ```isolcpus=managed_irq,domain,X-Y```

Reply:
Regarding kernel versions, I am all for testing everything and failing as early as possible, but my state of mind was the whole point of the way the system is made is to not have to check this stuff, meaning I hunted down the RHCOS kernel used for 4.5 and 4.6 *before* I started to work on PAO and the code was in.
So my questions are:
1. How can we get in a situation we have an older kernel (do we support at all?) ? 
2. Does it worth the check even if would happen for a not supported kernel and the worth case scenario is a dmesg info line? (not even a warning)
`  pr_info("isolcpus: Skipped unknown flag %.*s\n", len, par);`

If you still think is necessary, I'll start working on it.

Reply:
great that you mentioned the `domain` part, I was reading about that and I was wondering (and about to ask) why was it left out

Reply:
I added the  'domain' flag to the PR now.

Reply:
Ack, it makes much more sense now. But that brings me to another question. Should we use isolcpus=managed_irq,X (without domain) for the non-static case?

Reply:
Well, it seems it can't hurt, in non-static case we will have systemd.cpu_affinity=${not_isolated_cores_expanded}  which will kind of avoid the reserved cpus while  isolcpus=managed_irq,X will avoid triggering interrupts on them.

On the other hand in the static case is clear the user wants the reserved CPU completely isolated and it makes sense to not have IRQs triggered on them at the expense of interrupts starvation (e.g. net packet would be lost since we don't have enough CPUs to process the IRQs), however in the non-static case it possible the starvation may be an issue (theoretically).

After a session with Yanir we decided to consult with Andrew Theurer about this.

Reply:
Here is Andrew's answer from a private chat:

> whatever CPUs are in “isolated:” in the profile should be in  the managed_irq list
> basically any cpu that is in the list for rcu_nocbs should also be used for managed_irq

I'll get to it

Reply:
Done, repushed